### PR TITLE
Decorators

### PR DESCRIPTION
**Pull request by anas599**
In this PR:

- Create a class Nameable
- Implement a method called correct_name
- Prepare base Decorator
- Prepare CapitalizeDecorator: Implement a method that capitalizes the output of @nameable.correct_name.
- TrimmerDecorator: Implement a method correct_name that makes sure that the output of @nameable.correct_name

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @anas599, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a Live demo section in your README file, kindly make sure to remove this section altogether because your app is now a web app nor deployed

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: lib/person.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```
Comment (rloterh) [inline lib/person.rb:33]: - [ ] [OPTIONAL]
It's a best practice to mark `nameable` with `attr_accessor`, to give an indication that we intend to both _read_ and _write_ this variable. Let's endeavor to improve upon the same.

File: README.md [hunk 2]
```diff
@@ -43,9 +42,6 @@ PostgresSQL
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (rloterh) [inline README.md]: - [ ] [OPTIONAL]
Kindly ensure to streamline the highlighted portions of your ReadMe so as to make it stand out better!

image

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Anas,  

Your project is complete!🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Decorator pattern properly implemented!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- Kindly endeavor to remove the guide note below the MIT license.

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**